liquibase/src/test/java/org/openmrs/liquibase/SchemaOnlyTunerTest.java

```diff
@@ -0,0 +1,177 @@
+		// given
+		String sourcePath = PATH_TO_TEST_RESOURCES + LIQUIBASE_SCHEMA_ONLY_SNAPSHOT_XML;
+		String targetPath = tempFolder.newFile("liquibase-schema-only-UPDATED-SNAPSHOT.xml").getPath();
+		
+		// when
+		schemaOnlyTuner.createUpdatedChangeLogFile(sourcePath, targetPath);
+		
+		// then
+		Document expected = schemaOnlyTuner.readChangeLogResource(LIQUIBASE_SCHEMA_ONLY_UPDATED_SNAPSHOT_XML);
+		Document actual = schemaOnlyTuner.readChangeLogFile(targetPath);
+		
+		assertThat(expected.asXML(), equalToCompressingWhiteSpace(actual.asXML()));
+	}
+	
+	@Test
+	public void shouldDetachChangeSetsForLiquibaseTables() {
+		// given
+		XPath xpathOne = DocumentHelper.createXPath("//dbchangelog:createTable[@tableName=\"liquibasechangelog\"]\"");
+		xpathOne.setNamespaceURIs(namespaceUris);
+		
+		XPath xpathTwo = DocumentHelper.createXPath("//dbchangelog:createTable[@tableName=\"liquibasechangeloglock\"]\"");
+		xpathTwo.setNamespaceURIs(namespaceUris);
+		
+		assertEquals(1, xpathOne.selectNodes(document).size());
+		assertEquals(1, xpathTwo.selectNodes(document).size());
+		
+		// when
+		Document actual = schemaOnlyTuner.detachLiquibaseTables(document);
+		
+		// then
+		assertEquals(0, xpathOne.selectNodes(actual).size());
+		assertEquals(0, xpathTwo.selectNodes(actual).size());
+	}
+	
+	@Test
+	public void shouldReplaceBitTypeWithBooleanType() {
+		// given
+		XPath xpath = DocumentHelper.createXPath("//dbchangelog:column[@type=\"BIT(1)\"]/attribute::type");
+		xpath.setNamespaceURIs(namespaceUris);
+		
+		List<Node> nodes = xpath.selectNodes(document);
+		assertEquals(94, nodes.size());
+		
+		// when
+		Document actual = schemaOnlyTuner.replaceBitWithBoolean(document);
+		
+		// then
+		for (Node node : nodes) {
+			assertEquals("BOOLEAN", node.getParent().attributeValue("type"));
+		}
+	}
+	
+	@Test
+	public void shouldReplaceLongtextTypeWithClobType() {
+		// given
+		XPath xPath = DocumentHelper.createXPath("//dbchangelog:column[@type=\"LONGTEXT\"]/attribute::type");
+		xPath.setNamespaceURIs(namespaceUris);
+		
+		List<Node> nodes = xPath.selectNodes(document);
+		assertEquals(1, nodes.size());
+		
+		// when
+		SchemaOnlyTuner schemaOnlyTunerSpy = Mockito.spy(schemaOnlyTuner);
+		Document actual = schemaOnlyTunerSpy.replaceLongtextWithClob(document);
+		
+		// then
+		for (Node node : nodes) {
+			assertEquals("CLOB", node.getParent().attributeValue("type"));
+		}
+		
+		Mockito.verify(schemaOnlyTunerSpy).assertLongtextNodes(any());
+	}
+	
+	@Test
+	public void shouldAssertLongtextNode() {
+		// given
+		XPath xPath = DocumentHelper.createXPath("//dbchangelog:column[@type=\"LONGTEXT\"]/attribute::type");
+		xPath.setNamespaceURIs(namespaceUris);
+		
+		List<Node> nodes = xPath.selectNodes(document);
+		assertEquals(1, nodes.size());
+		
+		// when and then
+		assertTrue(schemaOnlyTuner.assertLongtextNodes(nodes));
+	}
+	
+	@Test(expected = AssertionError.class)
```

Comment:
thesting exceptions is done only with assertThrows in JUnit 5

Reply:
Done.